Fixing visible message being visible even if the mob is in the object

HMBGERDO opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## What Does This PR Do
<!-- Include a small to medium description of what your PR changes. Document all changes, as not doing this may delay reviews or even discourage maintainers from merging your PR! -->
If mob is covered with a locker or other object we shouldnt see their visible messages. This PR changes mob.visible_message logic to check if mob.loc is turf. If its not, we check if other mob in the same object. If they are in the same object, show message. If they are not but next to it, we show blind_message(if we have this param setted)

## Why It's Good For The Game
<!-- Add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->
You shouldnt see someone preparing their krav maga gloves attack or anything else if they are in locker or something. You may want to still see someone doing that in glass objects like cryotube or anything else but i dont think we have a lot of these objects being really used in game. This PR makes your visible messages really VISIBLE, hiding em if you are in the other object

## Testing
<!-- How did you test the PR, if at all? -->
compiled and runned on current build

## Changelog
:cl:
fix: Makes visible messages visible only if mob is on the turf or in the same object as you
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->
<!-- If a PR has no impact on players (i.e. a code refactor that does not change functionality) then the entire Changelog heading and contents can be removed. -->

Sirryan2002 commented on code/modules/mob/mob.dm:
```diff
@@ -106,6 +106,11 @@
 // blind_message (optional) is what blind people will hear e.g. "You hear something!"
 
 /mob/visible_message(message, self_message, blind_message)
+	if(!isturf(src.loc)) // cant see what do mob do in a locker or something
```
```suggestion
	if(!isturf(loc)) //mobs inside objects (such as lockers) shouldn't have their actions visible to those outside the object
```
src is implied here.  Also your comment needs to be more clear

HMBGERDO replied:
removed in update
comment changed in update

Sirryan2002 commented on code/modules/mob/mob.dm:
```diff
@@ -106,6 +106,11 @@
+	if(!isturf(src.loc)) // cant see what do mob do in a locker or something
+		if(self_message)
+			var/msg = self_message
+			src.show_message(msg, EMOTE_VISIBLE, blind_message, EMOTE_AUDIBLE)
```
```suggestion
			show_message(msg, EMOTE_VISIBLE, blind_message, EMOTE_AUDIBLE)
```
src is implied

HMBGERDO replied:
removed in update

Sirryan2002 commented on code/modules/mob/mob.dm:
```diff
@@ -106,6 +106,11 @@
 /mob/visible_message(message, self_message, blind_message)
+	if(!isturf(src.loc)) // cant see what do mob do in a locker or something
+		if(self_message)
+			var/msg = self_message
```
there's no point to defining `msg` here if you use it once and are assigning an already available parameter `self_message` without modifying it.

HMBGERDO replied:
changed it in update

Sirryan2002 reviewed (changes requested):
My concern with this check here is that any mob that is also in the object with the src mob should be able to see this message but cannot since this check limits the visible message to only the src mob anytime it's loc is not a turf.

Additionally, visible message has a `blind_message` parameter (why this isn't just limited to `audible_message()`, I have no idea) which sends messages that indicate to the listener what they're hearing. Being a closet, crate, cryo tube, sleeper, etc will not silence noises realistically and won't suppress sound files being played so it seems odd that in this case the visible message will be completely unheard by anyone nearby. Perhaps if a mob is in an object, nearby people in the object can only hear the blind_message?

HMBGERDO commented:
fixed all implied src, changed logic to show visible message if you are in the same object, show blind_message if we have one and you are next to this object(in 3 view)
added invisible check

Sirryan2002 commented on code/modules/mob/mob.dm:
```diff
@@ -106,6 +106,19 @@
+			var/msg = message
+			if(self_message && M == src)
+				msg = self_message
+			if(!(M.loc == loc))
```
```suggestion
			if(M.loc != loc)
```
one final nitpick on my end
